lib/internal/loader/CreateDynamicModule.js

```diff
@@ -5,7 +5,7 @@ const debug = require('util').debuglog('esm');
 const ArrayJoin = Function.call.bind(Array.prototype.join);
 const ArrayMap = Function.call.bind(Array.prototype.map);
 
-const createDynamicModule = (exports, url = '', evaluate) => {
+function createDynamicModule(exports, url = '', evaluate) {
```

Comment:
please change this back to an arrow function, it lacks a `.prototype` and protects against `new`

Reply:
will do, also did you see https://github.com/nodejs/node/pull/16874#issuecomment-343013291

Reply:
I did, but this PR is a lot of changes which makes me a bit slow to re-review so quickly